stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsListFragment.kt

```diff
@@ -24,7 +26,23 @@ internal class PaymentOptionsListFragment(
 
     override val sheetViewModel: PaymentOptionsViewModel by lazy { activityViewModel }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        // The onViewCreated will set the default selected item on the list, and this
+        // will replace whatever was in the last selection.  However, we need to keep
+        // this selected item if it was a new card, so that we transition to the last
+        // unsaved card details.
+        val lastNewCard = sheetViewModel.newCardSelection
```

Comment:
This seems like a risky pattern. Can we move this logic into `BasePaymentMethodsListFragment#onViewCreated()` or the ViewModel?

Reply:
yeah, I agree, I didn't really like this, I am still brainstorming on it.

Reply:
Just needed to make sure the adapter didn't effectively update the selection, unless the user actually clicked on a card to select it.